scp.c

```diff
@@ -976,19 +1044,37 @@ toremote(int argc, char **argv)
 			}
 			addargs(&alist, "--");
 			addargs(&alist, "%s", host);
-			addargs(&alist, "%s", cmd);
+			/* This will work only if the first remote scp supports sftp mode */
+			addargs(&alist, "%s%s", cmd, mode == MODE_SFTP ? " -M sftp" : "");
```

Comment:
Will this work? I thought it would treat "cmd -M sftp" as a single argument, though scp's horrible argument passing might avoid that.

Unless there's some reason to do it on a single line, an explicit "if (mode == MODE_SFTP) addargs(...)" would be better IMO

Reply:
Thank you. Indeed you are right. The arguments should come separately. Fixed now.